tools/fuzzer/platform/imx8-host.c

```diff
@@ -0,0 +1,515 @@
+#define ADSP_IMX8_HOST_IRAM_BASE        0x596f8000
+#define ADSP_IMX8_HOST_DRAM_BASE        0x596e8000
+#define SDRAM0_BASE     0x92400000
+#define SDRAM0_SIZE     0x800000
+#define SDRAM1_BASE     0x92C00000
+#define SDRAM1_SIZE     0x800000
+
+/* Mailbox configuration */
+#define ADSP_SRAM_OUTBOX_BASE       ADSP_IMX8_DSP_MAILBOX_BASE
+#define ADSP_SRAM_OUTBOX_SIZE       0x1000
+#define ADSP_SRAM_OUTBOX_OFFSET     0
+
+#define ADSP_SRAM_INBOX_BASE        (ADSP_SRAM_OUTBOX_BASE \
+					+ ADSP_SRAM_OUTBOX_SIZE)
+#define ADSP_SRAM_INBOX_SIZE        0x1000
+#define ADSP_SRAM_INBOX_OFFSET      ADSP_SRAM_OUTBOX_SIZE
+
+#define ADSP_SRAM_DEBUG_BASE        (ADSP_SRAM_INBOX_BASE \
+				     + ADSP_SRAM_INBOX_SIZE)
+#define ADSP_SRAM_DEBUG_SIZE        0x800
+#define ADSP_SRAM_DEBUG_OFFSET      (ADSP_SRAM_INBOX_OFFSET \
+				     + ADSP_SRAM_INBOX_SIZE)
+
+#define ADSP_SRAM_EXCEPT_BASE       (ADSP_SRAM_DEBUG_BASE \
+				     + ADSP_SRAM_DEBUG_SIZE)
+#define ADSP_SRAM_EXCEPT_SIZE       0x800
+#define ADSP_SRAM_EXCEPT_OFFSET     (ADSP_SRAM_DEBUG_OFFSET \
+				     + ADSP_SRAM_DEBUG_SIZE)
+
+#define ADSP_SRAM_STREAM_BASE       (ADSP_SRAM_EXCEPT_BASE \
+				     + ADSP_SRAM_EXCEPT_SIZE)
+#define ADSP_SRAM_STREAM_SIZE       0x1000
+#define ADSP_SRAM_STREAM_OFFSET     (ADSP_SRAM_EXCEPT_OFFSET \
+				     + ADSP_SRAM_EXCEPT_SIZE)
+
+#define ADSP_SRAM_TRACE_BASE        (ADSP_SRAM_STREAM_BASE \
+				     + ADSP_SRAM_STREAM_SIZE)
+#define ADSP_SRAM_TRACE_SIZE        0x1000
+#define ADSP_SRAM_TRACE_OFFSET      (ADSP_SRAM_STREAM_OFFSET \
+				     + ADSP_SRAM_STREAM_SIZE)
+
+#define ADSP_IMX8_DSP_MAILBOX_SIZE  (ADSP_SRAM_INBOX_SIZE \
+				     + ADSP_SRAM_OUTBOX_SIZE \
+				     + ADSP_SRAM_DEBUG_SIZE \
+				     + ADSP_SRAM_EXCEPT_SIZE \
+				     + ADSP_SRAM_STREAM_SIZE \
+				     + ADSP_SRAM_TRACE_SIZE)
+#define ADSP_IMX8_IRAM_SIZE          0x8000
+#define ADSP_IMX8_DRAM_SIZE          0x8000
+
+#define ADSP_IMX8_DSP_MU_SIZE         0x10000
+#define XSHAL_MU13_SIDEB_BYPASS_PADDR 0x5D310000
+#define ADSP_IMX8_DSP_MUB_BASE         XSHAL_MU13_SIDEB_BYPASS_PADDR
+#define ADSP_IMX8_DSP_MUA_BASE		0x5D280000
+
+#define ADSP_MAILBOX_SIZE			0x1000
+
+/* TODO get from driver. */
+#define IMX8_PANIC_OFFSET(x)	(x)
+
+struct imx8_data {
+	void *bar[MAX_BAR_COUNT];
+	struct mailbox host_box;
+	struct mailbox dsp_box;
+	int boot_complete;
+	pthread_mutex_t mutex;
+};
+
+/* Platform host description taken from Qemu
+ * - mapped to BAR 0, 1, 2, 3
+ */
+static struct fuzzer_mem_desc imx8_mem[] = {
+	{.name = "iram", .base = ADSP_IMX8_HOST_IRAM_BASE,
+	 .size = ADSP_IMX8_IRAM_SIZE},
+	{.name = "dram", .base = ADSP_IMX8_HOST_DRAM_BASE,
+	 .size = ADSP_IMX8_DRAM_SIZE},
+	{.name = "sdram0", .base = SDRAM0_BASE,
+	 .size = SDRAM1_SIZE},
+	{.name = "sdram1", .base = SDRAM1_BASE,
+	 .size = SDRAM1_SIZE},
+};
+
+/* mapped to BAR 4, 5, 6 */
+static struct fuzzer_reg_space imx8_io[] = {
+	{ .name = "mu_a",
+	  .desc = {.base = ADSP_IMX8_DSP_MUA_BASE,
```

Comment:
Lets call this mu13_a, mu13_b. On i.MX8/8X DSP communicates with Host processor via MU13.